packages/dynamodb/src/index.ts

```diff
@@ -1,447 +1,436 @@
+
+      const data = await client.update({
+        TableName,
+        Key: {
+          // next-auth type is incorrect it should be Partial<AdapterUser> & {id: string} instead of just Partial<AdapterUser>
+          pk: `USER#${user.id as string}`,
+          sk: `USER#${user.id as string}`,
         },
-        async unlinkAccount(userId, providerId, providerAccountId) {
-          const deleted = await client
-            .delete({
-              TableName,
-              Key: {
-                pk: `USER#${userId}`,
-                sk: `ACCOUNT#${providerId}#${providerAccountId}`,
-              },
-            })
-            .promise()
+        UpdateExpression: updateExpression,
+        ExpressionAttributeNames: ExpressionAttributeNames,
+        ExpressionAttributeValues: ExpressionAttributeValues,
+        ReturnValues: "ALL_NEW",
+      })
+      const attributesDynamo = data.Attributes as AdapterUserDynamo
+      const attributes = unMarshallAdapterUserDynamo(attributesDynamo)
+      return attributes
+    },
+    async deleteUser(userId) {
+      // query all the items related to the user to delete
+      const resQuery = await client.query({
+        TableName,
+        KeyConditionExpression: "#pk = :pk",
+        ExpressionAttributeNames: { "#pk": "pk" },
+        ExpressionAttributeValues: { ":pk": `USER#${userId}` },
+      })
+      if (!resQuery.Items) {
+        return null
+      }
+      const items = resQuery.Items as Array<
+        AdapterUserDynamo | AdapterSessionDynamo
+      >
+      // find the user we want to delete to return at the end of the function call
+      const user = items.find((item) => item.type === "USER") as
+        | AdapterUserDynamo
+        | undefined
+      const itemsToDelete = items.map((item) => {
+        return {
+          DeleteRequest: {
+            Key: {
+              sk: item.sk,
+              pk: item.pk,
+            },
+          },
+        }
+      })
+      // batch write commands cannot handle more than 25 requests at once
+      const itemsToDeleteMax = itemsToDelete.slice(0, 25)
+      const param: BatchWriteCommandInput = {
+        RequestItems: { [TableName]: itemsToDeleteMax },
+      }
+      await client.batchWrite(param)
+      return user ? unMarshallAdapterUserDynamo(user) : null
+    },
+    async linkAccount(data) {
+      const item: Account = {
+        pk: `USER#${data.userId}`,
+        sk: `ACCOUNT#${data.provider}#${data.providerAccountId}`,
+        GSI1PK: `ACCOUNT#${data.provider}`,
+        GSI1SK: `ACCOUNT#${data.providerAccountId}`,
+        userId: data.userId,
+        provider: data.provider,
+        providerAccountId: data.providerAccountId,
+        type: data.type,
+        accessToken: data.access_token,
+        expiresAt: data.expires_at,
+        idToken: data.id_token,
+        refreshToken: data.refresh_token,
+        tokenType: data.token_type,
+        scope: data.scope,
+        sessionState: data.session_state,
+      }
```

Comment:
```suggestion
      const item: Account = {
        pk: `USER#${data.userId}`,
        sk: `ACCOUNT#${data.provider}#${data.providerAccountId}`,
        GSI1PK: `ACCOUNT#${data.provider}`,
        GSI1SK: `ACCOUNT#${data.providerAccountId}`,
        userId: data.userId,
        provider: data.provider,
        providerAccountId: data.providerAccountId,
        type: data.type,
        accessToken: data.access_token,
        expiresAt: data.expires_at,
        idToken: data.id_token,
        refreshToken: data.refresh_token,
        tokenType: data.token_type,
        scope: data.scope,
        sessionState: data.session_state,
      }
```

This should be much more dynamic. Use the `...` spread operator to don't lock the adapter in on certain fields.

Reply:
I removed the code to transform snake case to camel case, dyanmodb doesn't care about the case for fields.